src/libstore/local-store.cc

```diff
@@ -136,7 +136,9 @@ LocalStore::LocalStore(
     for (auto & perUserDir : {profilesDir + "/per-user", gcRootsDir + "/per-user"}) {
         createDirs(perUserDir);
         if (!readOnly) {
-            if (chmod(perUserDir.c_str(), 0755) == -1)
+            struct stat st;
+            stat(perUserDir.c_str(), &st);
+            if (!(st.st_mode & 0755) && chmod(perUserDir.c_str(), 0755) == -1)
```

Comment:
This needs a comment about the intended behaviour. This will skip the chmod call if any of the bits in 0755 are already set? If so I don't really understand the motivation for that.

Reply:
The intended way is that it skips the chmod call if st_mode is 0755